index.js

```diff
@@ -158,6 +158,36 @@ Object.defineProperty(File.prototype, 'relative', {
   }
 });
 
+Object.defineProperty(File.prototype, 'dirname', {
```

Comment:
dirname has a different meaning and can actually _be_ different than file.base. For example:

``` js
gulp.src('assets/css/**/*.scss')
```

Every file's file.base will be `assets/css` but any file in a subdirectory will have a different dirname.